Classes/Emogrifier.php

```diff
@@ -323,6 +334,81 @@ protected function process(\DOMDocument $xmlDocument)
     }
 
     /**
+     * Applies $newStyles to $node.
+     *
+     * This method maps CSS styles to HTML attributes and adds those to the
+     * node.
+     *
+     * @param string[] $newStyles the new CSS styles taken from the global styles
+     *                            to be applied to this node.
+     * @param \DOMNode $node      node to apply styles to.
+     *
+     * @return void
+     */
+    private function mapStyles2HtmlAttrs($newStyles, \DOMNode $node)
+    {
+        foreach ($newStyles as $styleName => $styleValue) {
+            // Strip !important indicator
+            $styleValue = str_replace('!important', '', $styleValue);
+            if ($styleName == 'background-color') {
+                // Always override HTML attributes. CSS takes precedence.
+                $node->setAttribute('bgcolor', $styleValue);
+            } elseif ($styleName == 'background') {
+                // Parse out the color, if any
+                $styles = explode(' ', $styleValue);
+                $first = $styles[0];
+                if (!is_numeric(substr($first, 0, 1)) && substr($first, 0, 3) != 'url') {
```

Comment:
is_numeric tests a single digit, so I'm not worried about other formats. Basically, it needs to test whether the first background style is a color. If the property starts with a digit, the color is omitted and it is a position. If the property starts with url, the color is omitted and it's an image.

I should add checks for repeat and named positions.